lib/ObstacleHandler.cpp

```diff
@@ -85,7 +85,7 @@ ObstacleInfo * ObstacleHandler::loadFromJson(const std::string & scope, const Js
 	info->animation = json["animation"].String();
 	info->width = json["width"].Integer();
 	info->height = json["height"].Integer();
-	for(auto & t : json["allowedTerrain"].Vector())
+	for(auto & t : json["allowedTerrains"].Vector())
```

Comment:
If you rename it here, you should also rename it in all mods, which already use "allowedTerrain".
AFAIK there are 2 mods: hota and forge, change is needed in obstacles.json for vcmi-1.1 branches

Reply:
Ok, created 2 more pull requests for hota and forge. Directly in vcmi-1.1 branches.